graphql_compiler/schema_generation/sqlalchemy/utils.py

```diff
@@ -4,15 +4,22 @@
 
 def validate_that_tables_have_primary_keys(tables):
     """Validate that each SQLAlchemy Table object has a primary key."""
+    tables_missing_primary_keys = set()
     for table in tables:
         if not table.primary_key:
-            raise MissingPrimaryKeyError(
-                "SQLAlchemy Table with name {} and schema {} is missing a "
-                "primary key. Note that the primary keys in SQLAlchemy "
-                "Table objects do not have to match the primary keys in "
-                "the underlying row. They must simple be unique and "
-                "non-null identifiers of each row.".format(table.name, table.schema)
-            )
+            tables_missing_primary_keys.add(table)
+    if tables_missing_primary_keys:
+        error_message = (
+            "At least one SQLAlchemy Table is missing a "
+            "primary key. Note that the primary keys in SQLAlchemy "
+            "Table objects do not have to match the primary keys in "
+            "the underlying row. They must simply be unique and "
+            "non-null identifiers of each row. The tables missing primary "
+            "keys have names and schemas as follows: "
+        )
+        for table in tables_missing_primary_keys:
+            error_message += "name: {} schema: {} ".format(table.name, table.schema)
```

Comment:
Use `" ".join(<stuff>)` instead of `error_message +=` — the `+=` approach ends up being the Shlemiel the Painter's algorithm: https://www.joelonsoftware.com/2001/12/11/back-to-basics/